### PR TITLE
Added the CLI steps in provisioning guide

### PR DESCRIPTION
**Pull request by Marekys**
Added CLI steps for these guides: 
creating compute profiles
adding amazon ec2 details to a compute profile 
adding openstack details to a compute profile

* [yes] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:
(Im not sure what this means)
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

### CODE REVIEW

**Comment by Marekys** [comment]
This is the issue this PR refers to: 
https://github.com/theforeman/foreman-documentation/issues/459

File: guides/common/modules/proc_adding-amazon-ec2-details-to-a-compute-profile.adoc
Comment (asteflova) [inline guides/common/modules/proc_adding-amazon-ec2-details-to-a-compute-profile.adoc:18]: Us docs people are huge fans of consistency :) Notice how at some places you're using the wording "use the following command" and in others "enter the following command". It's not a big issue, but I still can't help pointing it out.

Comment (asteflova) [reply guides/common/modules/proc_adding-amazon-ec2-details-to-a-compute-profile.adoc:18]: Writing out "use the following command" does not add any value; users can see that the whole step is about using a command. A more helpful way to document commands is to describe what those commands do. A good example of this approach is section [2.12. Creating an installation medium for Ubuntu 22.04](https://theforeman-foreman-documentation-preview-pr-2988.surge.sh/nightly/Provisioning_Hosts/index-foreman-el.html#Creating_an_Installation_Medium_for_Ubuntu_22_04_provisioning): notice how each of the steps in that procedure talk about _what_ that command does while avoiding words like "use this command to [...]".

Can you think of a way to rephrase line 18 (and the other steps in your PR) to be more descriptive like this?

File: guides/common/modules/proc_creating-compute-profiles.adoc
Comment (asteflova) [inline guides/common/modules/proc_creating-compute-profiles.adoc:5]:
```suggestion

To use the CLI instead of the {ProjectWebUI}, see the xref:cli-creating-compute-profiles_{context}[].

```
Most of the other occurrences of this sentence in the guide use a separate paragraph. Also, the plus sign at the end seems like it should not be there.

**Review by asteflova — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_adding-openstack-details-to-a-compute-profile.adoc
Comment (ekohl) [inline guides/common/modules/proc_adding-openstack-details-to-a-compute-profile.adoc:25]: Elsewhere we use the `{OpenStack}` attribute, which for Red Hat uses the brand name. Otherwise it also needs to the correct casing. So either
```suggestion
. To add OpenStack details to a compute profile, use the following command:
```

Or
```suggestion
. To add {OpenStack} details to a compute profile, use the following command:
```

**Review by ekohl — COMMENTED** [review]
I added the issue to this PR so that once this is merged, the issue is closed. If that wasn't the intention, then please let us know.

File: guides/common/modules/proc_creating-compute-profiles.adoc
Comment (asteflova) [inline guides/common/modules/proc_creating-compute-profiles.adoc:35]:
```suggestion
. Set attributes for the compute profile:
```

Comment (asteflova) [reply guides/common/modules/proc_creating-compute-profiles.adoc:35]: Just to explain this suggestion:

* s/atributes/attributes/ -- typo
* s/using// -- redundant, and also we have a convention to avoid using incomplete sentences in introduction like these (incomplete sentence might cause problems when translating).
* s/Compute Profile/compute profile/ -- the existing content uses lower case (see for example the heading on line 2)

Comment (asteflova) [reply guides/common/modules/proc_creating-compute-profiles.adoc:35]: Please check the rest of your PR and make sure you're using `compute profile` rather than `Compute Profile` everywhere in your additions.

BTW, don't let line 20 puzzle you:

> . In the {ProjectWebUI}, navigate to *Infrastructure* > *Compute Profiles* and click *Create Compute Profile*.

This one uses `Compute Profile` in upper case formatting and that's okay (sort of) because that's what the labels in the Foreman web UI say.

Screenshot from 2024-04-30 19-27-39

The documentation writes this differently, which is not ideal, but it's also out of scope of your PR so you can just stick to the documentation conventions in your PR.

**Review by asteflova — CHANGES_REQUESTED** [review]
Thanks for implementing the feedback so far. I have a couple of more suggestions for you.

Also, I noticed one discrepancy that I find quite significant: The section in proc_creating-compute-profiles.adoc is named "Creating compute profiles". But quite a few of the commands you're adding are _not_ related to creating compute profiles: deleting a profile, viewing the details of a profile... Please take a closer look at what you're adding in that file and think about the intended scope of that section.

One way to do that would be to take a look at the web UI procedure above (lines 19+): Think about the steps described for the web UI and limit your CLI addition to mirror the web UI steps.

File: guides/common/modules/proc_creating-compute-profiles.adoc
Comment (Marekys) [reply guides/common/modules/proc_creating-compute-profiles.adoc:35]: Thanks for the review.

I was quite unsure if i should add the delete, view details commands. My thought was that those commands are nowhere in the documentation and it would be quite useful for the user to know that those commands exist and how to use them. But i will try to only keep the commands that are the same type as the procedure above as you mentioned. 

That typo was silly mistake, will quickly fix that, same about "using", i was looking through the docs and saw that "Compute Profiles" is used way more than "compute profiles" so thats why i decided for the first option. Looking back i can see the reasons why are which versions used, the reason being the UI reference.

Going to fix it asap and push new version.

File: guides/common/modules/proc_creating-compute-profiles.adoc
Comment (asteflova) [inline guides/common/modules/proc_creating-compute-profiles.adoc:46]:
```suggestion
. Optional: To update the attributes of a compute profile, specify the attributes you want to change.
For example, to change the number of CPUs and memory size:
```
This is mostly just about the conventions for the Foreman project (for example, we use the 'one sentence per line convention', so I added a new line).

File: guides/common/modules/proc_creating-compute-profiles.adoc
Comment (asteflova) [reply guides/common/modules/proc_creating-compute-profiles.adoc:43]: Italics for a replaceable value.

File: guides/common/modules/proc_creating-compute-profiles.adoc
Comment (asteflova) [inline guides/common/modules/proc_creating-compute-profiles.adoc:58]:
```suggestion
. Optional: To change the name of the compute profile, use the `--new-name` attribute:
```

**Review by asteflova — CHANGES_REQUESTED** [review]
Just a few minor suggestions, mostly to apply this project's specific documentation conventions.

**Comment by asteflova** [comment]
> I was quite unsure if i should add the delete, view details commands. My thought was that those commands are nowhere in the documentation and it would be quite useful for the user to know that those commands exist and how to use them.

@Marekys That was a reasonable thing to consider. My preferred approach in these cases is to link to in-product documentation (man pages, command --help texts). Because when you list commands in a guide just because users might not know about them, you're also basically duplicating the information that's already available in the in-product documentation. (To be completely honest, other writers' opinions may vary, though :) )

If you really want to let users know that these commands exist, you could also point them to these in-product docs. Take a look at [section Creating a User](https://docs.theforeman.org/nightly/Administering_Project/index-katello.html#Creating_a_User_admin). That section has a segment named [Additional resources](https://github.com/theforeman/foreman-documentation/blob/d570d3098edc61e8535cca641d881b0ea18d31d8/guides/common/modules/proc_creating-a-user.adoc?plain=1#L47C1-L49C3); you could use a segment like that in your section to point users to `hammer compute-profile --help`.

```
# hammer compute-profile --help
Usage:
    hammer compute-profile [OPTIONS] SUBCOMMAND [ARG] ...

Parameters:
 SUBCOMMAND                    Subcommand
 [ARG] ...                     Subcommand arguments

Subcommands:
 create                        Create a compute profile
 delete                        Delete a compute profile
 info                          Show a compute profile
 list                          List of compute profiles
 update                        Update a compute profile
 values                        Create update and delete Compute profile values

Options:
 -h, --help                    Print help
```

**Comment by Marekys** [comment]
@asteflova Thanks for the suggestions.

Silly mistakes but important, made all the suggestions also decided to add the "Additional resources" section as i find it quite helpful.

Also there was one unnecessary attribute in one of the commands which i could not find anywhere else in either documentation, probably because it was in the old implementation.

**Review by asteflova — CHANGES_REQUESTED** [review]
Just a few more suggestions, all should be easy to apply.

When you're done, please also resolve comment threads with feedback you have already applied (see the `Resolve conversation` button under each thread).

**Comment by asteflova** [comment]
Thanks for all your work so far @Marekys, I like where this is going. I believe that after you apply the latest batch of feedback from me, I'll be able to approve this PR, and then we can wait for a technical review from @stejskalleos.

File: guides/common/modules/proc_creating-compute-profiles.adoc
Comment (Marekys) [reply guides/common/modules/proc_creating-compute-profiles.adoc:46]: Yes i did, thanks for mentioning it

**Comment by Marekys** [comment]
@asteflova  Applied the requested changes, hopefully correctly. Thanks for the fast responses, I really appreciate this. I hope we can finish this soon.

**Review by stejskalleos — APPROVED** [review]

**Review by asteflova — APPROVED** [review]

**Comment by asteflova** [comment]
Merged to "master" and cherry-picked:

* 035db009bc..0c2f699bf3  3.10 -> 3.10
* fe6362cac0..75ebe928d6  3.9 -> 3.9
* 499ef8d89b..2287db8423  3.8 -> 3.8
* 2c0b30e792..e7fdf7e208  3.7 -> 3.7
* 78b511513f..596cb045af  3.6 -> 3.6

**Comment by asteflova** [comment]
Thanks @Marekys and congratulations on your first contribution!

**Pull request merged**
